spec/workspace-spec.js

```diff
@@ -0,0 +1,2072 @@
+'use strict'
+
+/* global advanceClock, HTMLElement, waits */
+
+const path = require('path')
+const temp = require('temp').track()
+const TextEditor = require('../src/text-editor')
+const Workspace = require('../src/workspace')
+const Project = require('../src/project')
+const platform = require('./spec-helper-platform')
+const _ = require('underscore-plus')
+const fstream = require('fstream')
+const fs = require('fs-plus')
+
+describe('Workspace', () => {
+  let workspace
+  let setDocumentEdited
+
+  beforeEach(() => {
+    workspace = atom.workspace
+    workspace.resetFontSize()
+    spyOn(atom.applicationDelegate, 'confirm')
+    setDocumentEdited = spyOn(atom.applicationDelegate, 'setWindowDocumentEdited')
+    atom.project.setPaths([atom.project.getDirectories()[0].resolve('dir')])
+    waits(1)
+  })
+
+  afterEach(() => temp.cleanupSync())
+
+  describe('serialization', () => {
+    const simulateReload = () => {
+      const workspaceState = atom.workspace.serialize()
+      const projectState = atom.project.serialize({isUnloading: true})
+      atom.workspace.destroy()
+      atom.project.destroy()
+      atom.project = new Project({
+        notificationManager: atom.notifications,
+        packageManager: atom.packages,
+        confirm: atom.confirm.bind(atom),
+        applicationDelegate: atom.applicationDelegate
+      })
+      atom.project.deserialize(projectState)
+      atom.workspace = new Workspace({
+        config: atom.config,
+        project: atom.project,
+        packageManager: atom.packages,
+        grammarRegistry: atom.grammars,
+        deserializerManager: atom.deserializers,
+        notificationManager: atom.notifications,
+        applicationDelegate: atom.applicationDelegate,
+        viewRegistry: atom.views,
+        assert: atom.assert.bind(atom),
+        textEditorRegistry: atom.textEditors
+      })
+      return atom.workspace.deserialize(workspaceState, atom.deserializers)
+    }
+
+    describe('when the workspace contains text editors', () =>
```

Comment:
While you're here, could you make all the `describe` callbacks use curly braces? It can be kind of confusing when you go to add an `it` to some `describe` if the `describe` uses the single-expression syntax.